File: epicerie-vrac-projet-core/src/main/java/fr/epicerie_vrac_projet_core/models/Fournisseur.java
package fr.epicerie_vrac_projet_core.models;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.ManyToMany;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@Getter
@Setter
@ToString()
@Entity
public class Fournisseur extends Personne {
	
	private String nomEntreprise;
	private boolean producteur;
	private String description;
	
	@ManyToMany
	private Set<Article> articles = new HashSet<>();
	
}

File: epicerie-vrac-projet-angular/src/app/models/fournisseur.ts
import { Article } from "./article";
import { Personne } from "./personne";

export class Fournisseur {
    public id: number = 0;
    public nomEntreprise: string = "";
    public email: string = "";
    public producteur: boolean = true;
    public description: string = "";
    public personne: Personne = new Personne();
    public articles: Article[] = [];
}
